networkx/generators/tests/test_community.py

```diff
@@ -117,3 +117,19 @@ def test_gaussian_random_partition_graph():
     assert_raises(nx.NetworkXError,
                   nx.gaussian_random_partition_graph, 100, 101, 10, 1, 0)
 
+def test_ring_of_cliques():
+    for i in range(2, 20):
+        for j in range(1, 20):
+            G = nx.ring_of_cliques(i, j)
+            assert_equal(G.number_of_nodes(), i*j)
+            if i != 2 or j != 1:
+                assert_equal(G.number_of_edges(),
+                             len(list(itertools.
```

Comment:
Use the mathematical formula for the number of edges here instead of creating this list.